src/features/workspace/components/workspace-muxing-model.tsx

```diff
@@ -0,0 +1,269 @@
+import {
+  Alert,
+  Button,
+  Card,
+  CardBody,
+  CardFooter,
+  Form,
+  Input,
+  Label,
+  Link,
+  LinkButton,
+  Text,
+  TextField,
+  Tooltip,
+  TooltipInfoButton,
+  TooltipTrigger,
+} from "@stacklok/ui-kit";
+import { twMerge } from "tailwind-merge";
+import { useMutationPreferredModelWorkspace } from "../hooks/use-mutation-preferred-model-workspace";
+import {
+  MuxMatcherType,
+  V1ListAllModelsForAllProvidersResponse,
+} from "@/api/generated";
+import { FormEvent } from "react";
+import {
+  LayersThree01,
+  LinkExternal01,
+  Plus,
+  Trash01,
+} from "@untitled-ui/icons-react";
+import { SortableArea } from "@/components/SortableArea";
+import { WorkspaceModelsDropdown } from "./workspace-models-dropdown";
+import { useQueryListAllModelsForAllProviders } from "@/hooks/use-query-list-all-models-for-all-providers";
+import { useQueryMuxingRulesWorkspace } from "../hooks/use-query-muxing-rules-workspace";
+import {
+  PreferredMuxRule,
+  useMuxingRulesFormState,
+} from "../hooks/use-muxing-rules-form-workspace";
+import { FormButtons } from "@/components/FormButtons";
+
+function MissingProviderBanner() {
+  return (
+    // TODO needs to update the related ui-kit component that diverges from the design
+    <Alert
+      variant="warning"
+      className="bg-brand-200 text-primary font-normal dark:bg-[#272472]"
+      title="You must configure at least one provider before selecting your desired model."
+    >
+      <LinkButton
+        className="mt-4 text-white dark:bg-[#7D7DED]"
+        href="/providers"
+      >
+        Configure a provider
+      </LinkButton>
+    </Alert>
+  );
+}
+
+type SortableItemProps = {
+  index: number;
+  rule: PreferredMuxRule;
+  models: V1ListAllModelsForAllProvidersResponse;
+  isArchived: boolean;
+  showRemoveButton: boolean;
+  isDefaultRule: boolean;
+  setRuleItem: (rule: PreferredMuxRule) => void;
+  removeRule: (index: number) => void;
+};
+
+function SortableItem({
+  rule,
+  index,
+  setRuleItem,
+  removeRule,
+  models,
+  showRemoveButton,
+  isArchived,
+  isDefaultRule,
+}: SortableItemProps) {
+  const placeholder = isDefaultRule ? "Catch All" : "eg file type, file name";
+  return (
+    <div className="flex items-center gap-2" key={rule.id}>
+      <div className="flex w-full justify-between">
+        <TextField
+          aria-labelledby="filter-by-label-id"
+          onFocus={(event) => event.preventDefault()}
```

Comment:
:thinking:  why?

Reply:
I guess it is a leftover let me remove it